website/app/styles/pages/application/header.scss

```diff
@@ -1,84 +1,162 @@
 // HEADER
 
+@use "../../breakpoints" as breakpoint;
+@use "../../typography/mixins" as *;
+
+// notice: we use an ad-hoc breakpoint here to avoid overlapping of content
+$header-custom-breakpoint: 1000px;
+
 .doc-page-header {
-  // TBD with Heather if we want it sticky or not
   position: sticky;
   top: 0;
   // TODO review all the z-index stacking once the navigation has been finalized
   z-index: 2; // needs to be above the sidebar (when the content scrolls)
   display: flex;
   flex-direction: row;
   grid-area: header;
+  gap: 16px;
   align-items: center;
   justify-content: flex-start;
   height: var(--doc-page-header-height);
   padding: 0 24px;
-  color: #fff;
-  background: #000;
+  color: var(--doc-color-white);
+  background: var(--doc-color-black);
+
+  @media (min-width: $header-custom-breakpoint) {
+    gap: 32px;
+  }
 }
 
-// IDENTITY
+// LOGO
 
-.doc-page-header__identity {
-  display: grid;
-  grid-template-areas:
-    "logo name"
-    "logo version";
-  grid-template-rows: auto 1fr;
-  grid-template-columns: auto 1fr;
-  gap: 4px 8px;
-  align-items: center;
-}
+.doc-page-header__logo {
+  display: block;
+  flex: none;
+  height: 32px;
+  color: inherit;
+  text-decoration: none;
 
-.doc-page-header__identity-logo {
-  grid-area: logo;
+  &:hover,
+  &:active,
+  &:visited {
+    color: inherit;
+  }
 
-  img {
-    display: block;
-    width: 48px;
-    height: 48px;
+  svg {
+    width: auto;
+    height: 100%;
   }
-}
 
-.doc-page-header__identity-name {
-  grid-area: name;
-  color: #fff;
-  font-size: 16px;
-  font-family: sans-serif;
-  line-height: 1em;
-  text-decoration: none;
+  // we add a bit of "responsiveness" to the logo too, to gain some extra space :)
 
-  &:hover {
-    text-decoration: underline;
+  .doc-logo-hds__hashicorp {
+    @media (max-width: calc($header-custom-breakpoint - 1px)) {
+      display: none;
+    }
   }
 }
 
-.doc-page-header__identity-version {
-  grid-area: version;
-}
+// NAV MENU (PRIMARY)
 
-// MENU
+.doc-page-header__nav-menu {
+  display: none;
 
-.doc-page-header__top-menu {
-  margin-left: 40px;
+  @include breakpoint.medium-and-above() {
+    display: block;
+    flex: 1 0 auto;
+    align-self: stretch;
+  }
 
   ul {
     display: flex;
-    gap: 16px;
+    gap: 0;
+    align-items: center;
+    height: 100%;
     margin: 0;
     padding: 0;
     list-style: none;
+
+    @media (min-width: $header-custom-breakpoint) {
+      gap: 12px;
+    }
   }
+}
+
+.doc-page-header__nav-item-text {
+  align-self: stretch;
 
   a {
-    color: #fff;
-    font-size: 24px;
-    font-family: sans-serif;
-    line-height: 1em;
+    display: flex;
+    align-items: center;
+    height: 100%;
+    padding: 8px 12px 5px 12px; // we have to take in account the border-bottom
+    color: var(--doc-color-gray-400);
     text-decoration: none;
+    border-bottom: 3px solid transparent;
```

Comment:
The bottom border is from an outdated design. We moved away from it because it didn't provide the same level of impact after transitioning to the dark background. We also used this pattern too liberally across the navigational elements, and they started competing for attention. We've simplified things, so the activated page uses a slightly different color (`--doc-color-white`).

The most up-to-date designs can be found in [component handoffs](https://www.figma.com/file/Ky0qWjvHZR3je1lCBlzNB1/HDS-website?node-id=1345%3A79650&t=EgMPu6g8qiW7jKQw-4).